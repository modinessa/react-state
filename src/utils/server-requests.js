//const url = 'http://localhost:3000';
const url = '';

export function getUsers() {
  return fetch(`${url}/community`, {
    method: 'GET',
  });
}

export function getUser(id) {
	return fetch(`/community/${id}`);
}

export function subscribe(email) {
  return fetch(`${url}/subscribe`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      email: `${email}`,
    }),
  });
}

export function unsubscribe() {
  return fetch(`${url}/unsubscribe`, { method: 'POST' });
}
