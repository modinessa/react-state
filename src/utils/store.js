import { configureStore } from "@reduxjs/toolkit";
import usersReducer from "../components/reducer";

export const store = configureStore({
	reducer: {
		users: usersReducer,
	}
});