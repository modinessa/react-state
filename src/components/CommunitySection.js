import React, { useState, useEffect } from "react";
import { getUsers } from "../js/server-requests.js";
import { Users } from "./Users.js";

export function CommunitySection() {

	const [ isHide, setIsHide ] = useState(localStorage
				.getItem("IsHide") === "true" || false);
	const [ users, setUsers] = useState([]);
 
	useEffect(() => {
		getUsers()
		.then((response) => {
			return response.json()})
		.then(setUsers);
	}, []);

	return (
		<section className="app-section app-section--big-community">
			<div className={`app-section--big-community_title-block
											${isHide ? "add-margin" : ""}`}>
				<h2 className="app-title dark">
					Big Community of<br/>
					People Like You
				</h2>
				<button className="app-section--big-community_button"
					onClick={ () => {
						setIsHide(!isHide);
						localStorage.setItem("IsHide", !isHide);
					}}>
					{!isHide ? "Hide section" : "Show section"}
				</button>
			</div>
			{!isHide && 
				<>
					<h3 className="app-subtitle dark">
						We’re proud of our products, and we’re really excited<br/>
						when we get feedback from our users.
					</h3>
					<Users users={users} />
				</>}
		</section>
	)
}