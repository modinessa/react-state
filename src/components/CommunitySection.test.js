import {Provider} from "react-redux";
import renderer from 'react-test-renderer';
import {rest} from 'msw';
import {setupServer} from 'msw/node';
import {render, fireEvent} from '@testing-library/react';

import {store} from "../utils/store";
import {CommunitySection} from './CommunitySection';

const mockServer = setupServer(
	rest.get('/community', (req, res, ctx) => res(ctx.json([]))),
)

describe('CommunitySection', () => {
	it('renders component', () => {
		expect(renderer.create(
			<Provider store={store}>
				<CommunitySection/>
			</Provider>)).toMatchSnapshot();
	});

	it('hides users section', async () => {
		const communitySection = render(
			<Provider store={store}>
				<CommunitySection/>
			</Provider>);
		let hideButton = communitySection.getByLabelText('hide-users-section');
		
		expect(store.getState().users.isHide).toEqual(false);

		fireEvent.click(hideButton);

		hideButton = await communitySection.findByText('Show section');

		expect(hideButton).not.toBeUndefined();
		expect(store.getState().users.isHide).toEqual(true);

		fireEvent.click(hideButton);

		hideButton = await communitySection.findByText('Hide section');

		expect(hideButton).not.toBeUndefined();
		expect(store.getState().users.isHide).toEqual(false);
	});
});
