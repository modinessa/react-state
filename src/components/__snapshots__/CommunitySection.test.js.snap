// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunitySection renders component 1`] = `
<section
  className="app-section app-section--big-community"
>
  <div
    className="app-section--big-community_title-block
											"
  >
    <h2
      className="app-title dark"
    >
      Big Community of
      <br />
      People Like You
    </h2>
    <button
      aria-label="hide-users-section"
      className="app-section--big-community_button"
      onClick={[Function]}
    >
      Hide section
    </button>
  </div>
  <h3
    className="app-subtitle dark"
  >
    We’re proud of our products, and we’re really excited
    <br />
    when we get feedback from our users.
  </h3>
  <div
    className="app-section__users "
  />
</section>
`;
